diff --git a/CRMCall092016.xcodeproj/project.pbxproj b/CRMCall092016.xcodeproj/project.pbxproj
index 4695ff2..631b2f8 100644
--- a/CRMCall092016.xcodeproj/project.pbxproj
+++ b/CRMCall092016.xcodeproj/project.pbxproj
@@ -22,6 +22,8 @@
 		9266A7601DA5DFD70031082A /* NSDate+InternetDateTime.swift in Sources */ = {isa = PBXBuildFile; fileRef = 9266A75F1DA5DFD70031082A /* NSDate+InternetDateTime.swift */; };
 		92793B411D8F7FD000586A5F /* Cache.swift in Sources */ = {isa = PBXBuildFile; fileRef = 92793B401D8F7FD000586A5F /* Cache.swift */; };
 		9279E0901D9BA1B700F577A5 /* MainViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 9279E08E1D9BA1B700F577A5 /* MainViewController.swift */; };
+		928B488F1DB0D71100DBAAE3 /* StaffAvailabilityViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 928B488E1DB0D71100DBAAE3 /* StaffAvailabilityViewController.swift */; };
+		928B48911DB0DE2E00DBAAE3 /* StaffAvailabilityWindowController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 928B48901DB0DE2E00DBAAE3 /* StaffAvailabilityWindowController.swift */; };
 		92B0F0481D88E53700C70BC3 /* CRMCallHelpers.swift in Sources */ = {isa = PBXBuildFile; fileRef = 92B0F0471D88E53700C70BC3 /* CRMCallHelpers.swift */; };
 		92B2EC961D926F1800E487C9 /* CRMCallManager.swift in Sources */ = {isa = PBXBuildFile; fileRef = 92B2EC951D926F1800E487C9 /* CRMCallManager.swift */; };
 		92C261A21D8B85DF00413F54 /* Realm.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 92C261A01D8B85DF00413F54 /* Realm.framework */; };
@@ -85,6 +87,8 @@
 		9266A75F1DA5DFD70031082A /* NSDate+InternetDateTime.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = "NSDate+InternetDateTime.swift"; sourceTree = "<group>"; };
 		92793B401D8F7FD000586A5F /* Cache.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = Cache.swift; sourceTree = "<group>"; };
 		9279E08E1D9BA1B700F577A5 /* MainViewController.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = MainViewController.swift; sourceTree = "<group>"; };
+		928B488E1DB0D71100DBAAE3 /* StaffAvailabilityViewController.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = StaffAvailabilityViewController.swift; sourceTree = "<group>"; };
+		928B48901DB0DE2E00DBAAE3 /* StaffAvailabilityWindowController.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = StaffAvailabilityWindowController.swift; sourceTree = "<group>"; };
 		92B0F0471D88E53700C70BC3 /* CRMCallHelpers.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = CRMCallHelpers.swift; sourceTree = "<group>"; };
 		92B2EC951D926F1800E487C9 /* CRMCallManager.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = CRMCallManager.swift; sourceTree = "<group>"; };
 		92C261A01D8B85DF00413F54 /* Realm.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; path = Realm.framework; sourceTree = "<group>"; };
@@ -246,6 +250,7 @@
 				9279E08E1D9BA1B700F577A5 /* MainViewController.swift */,
 				925981BF1D9E68160006BE89 /* HistoryCallViewController.swift */,
 				92ECC33D1DAC727200A093EA /* CustomerListViewController.swift */,
+				928B488E1DB0D71100DBAAE3 /* StaffAvailabilityViewController.swift */,
 			);
 			path = ViewControllers;
 			sourceTree = "<group>";
@@ -286,6 +291,7 @@
 				925981C11D9E692C0006BE89 /* HistoryCallWindowController.swift */,
 				923BAAB81DA3388F00127BE8 /* RingIngWindowController.swift */,
 				92ECC3411DACA74D00A093EA /* CustomerListWindowController.swift */,
+				928B48901DB0DE2E00DBAAE3 /* StaffAvailabilityWindowController.swift */,
 			);
 			path = WindowController;
 			sourceTree = "<group>";
@@ -449,6 +455,7 @@
 				92C261A71D8BA54800413F54 /* RealmManager.swift in Sources */,
 				92B0F0481D88E53700C70BC3 /* CRMCallHelpers.swift in Sources */,
 				9258611D1D952A3300521D7D /* LoginViewController.swift in Sources */,
+				928B488F1DB0D71100DBAAE3 /* StaffAvailabilityViewController.swift in Sources */,
 				92E3E3131D8004C3008033ED /* AppDelegate.swift in Sources */,
 				92B2EC961D926F1800E487C9 /* CRMCallManager.swift in Sources */,
 				92E3E3291D8005E2008033ED /* AESExtension.m in Sources */,
@@ -468,6 +475,7 @@
 				925981C01D9E68160006BE89 /* HistoryCallViewController.swift in Sources */,
 				923BAAB91DA3388F00127BE8 /* RingIngWindowController.swift in Sources */,
 				92E3E3311D800947008033ED /* CRMCallConfig.swift in Sources */,
+				928B48911DB0DE2E00DBAAE3 /* StaffAvailabilityWindowController.swift in Sources */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 		};
diff --git a/CRMCall092016.xcworkspace/xcuserdata/hanbiro.xcuserdatad/UserInterfaceState.xcuserstate b/CRMCall092016.xcworkspace/xcuserdata/hanbiro.xcuserdatad/UserInterfaceState.xcuserstate
index 1c29533..7e180bf 100644
Binary files a/CRMCall092016.xcworkspace/xcuserdata/hanbiro.xcuserdatad/UserInterfaceState.xcuserstate and b/CRMCall092016.xcworkspace/xcuserdata/hanbiro.xcuserdatad/UserInterfaceState.xcuserstate differ
diff --git a/CRMCall092016.xcworkspace/xcuserdata/hanbiro.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist b/CRMCall092016.xcworkspace/xcuserdata/hanbiro.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist
index cd8707b..bfcb616 100644
--- a/CRMCall092016.xcworkspace/xcuserdata/hanbiro.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist
+++ b/CRMCall092016.xcworkspace/xcuserdata/hanbiro.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist
@@ -442,11 +442,123 @@
             ignoreCount = "0"
             continueAfterRunningActions = "No"
             filePath = "CRMCall092016/ViewControllers/CustomerListViewController.swift"
-            timestampString = "498044047.212306"
+            timestampString = "498119043.005097"
             startingColumnNumber = "9223372036854775807"
             endingColumnNumber = "9223372036854775807"
-            startingLineNumber = "143"
-            endingLineNumber = "143"
+            startingLineNumber = "170"
+            endingLineNumber = "170"
+            landmarkName = "searchCustomer()"
+            landmarkType = "5">
+         </BreakpointContent>
+      </BreakpointProxy>
+      <BreakpointProxy
+         BreakpointExtensionID = "Xcode.Breakpoint.FileBreakpoint">
+         <BreakpointContent
+            shouldBeEnabled = "Yes"
+            ignoreCount = "0"
+            continueAfterRunningActions = "No"
+            filePath = "CRMCall092016/ViewControllers/CustomerListViewController.swift"
+            timestampString = "498119043.005097"
+            startingColumnNumber = "9223372036854775807"
+            endingColumnNumber = "9223372036854775807"
+            startingLineNumber = "265"
+            endingLineNumber = "265"
+            landmarkName = "scrollRowToVisible(_:)"
+            landmarkType = "5">
+            <Locations>
+               <Location
+                  shouldBeEnabled = "Yes"
+                  ignoreCount = "0"
+                  continueAfterRunningActions = "No"
+                  symbolName = "CRMCall092016.CustomerListViewController.scrollRowToVisible (Swift.Int) -&gt; ()"
+                  moduleName = "CRMCall092016"
+                  usesParentBreakpointCondition = "Yes"
+                  urlString = "file:///Users/hanbiro/Desktop/PG/CRM_CALL/CRMCall092016/CRMCall092016/ViewControllers/CustomerListViewController.swift"
+                  timestampString = "498119981.616124"
+                  startingColumnNumber = "9223372036854775807"
+                  endingColumnNumber = "9223372036854775807"
+                  startingLineNumber = "265"
+                  endingLineNumber = "265"
+                  offsetFromSymbolStart = "42">
+               </Location>
+               <Location
+                  shouldBeEnabled = "Yes"
+                  ignoreCount = "0"
+                  continueAfterRunningActions = "No"
+                  symbolName = "CRMCall092016.CustomerListViewController.(scrollRowToVisible (Swift.Int) -&gt; ()).(implicit closure #1)"
+                  moduleName = "CRMCall092016"
+                  usesParentBreakpointCondition = "Yes"
+                  urlString = "file:///Users/hanbiro/Desktop/PG/CRM_CALL/CRMCall092016/CRMCall092016/ViewControllers/CustomerListViewController.swift"
+                  timestampString = "498119981.616385"
+                  startingColumnNumber = "9223372036854775807"
+                  endingColumnNumber = "9223372036854775807"
+                  startingLineNumber = "265"
+                  endingLineNumber = "265"
+                  offsetFromSymbolStart = "29">
+               </Location>
+            </Locations>
+         </BreakpointContent>
+      </BreakpointProxy>
+      <BreakpointProxy
+         BreakpointExtensionID = "Xcode.Breakpoint.FileBreakpoint">
+         <BreakpointContent
+            shouldBeEnabled = "Yes"
+            ignoreCount = "0"
+            continueAfterRunningActions = "No"
+            filePath = "CRMCall092016/ViewControllers/CustomerListViewController.swift"
+            timestampString = "498119043.005097"
+            startingColumnNumber = "9223372036854775807"
+            endingColumnNumber = "9223372036854775807"
+            startingLineNumber = "268"
+            endingLineNumber = "268"
+            landmarkName = "scrollRowToVisible(_:animated:)"
+            landmarkType = "5">
+            <Locations>
+               <Location
+                  shouldBeEnabled = "Yes"
+                  ignoreCount = "0"
+                  continueAfterRunningActions = "No"
+                  symbolName = "CRMCall092016.CustomerListViewController.scrollRowToVisible (Swift.Int, animated : Swift.Bool) -&gt; ()"
+                  moduleName = "CRMCall092016"
+                  usesParentBreakpointCondition = "Yes"
+                  urlString = "file:///Users/hanbiro/Desktop/PG/CRM_CALL/CRMCall092016/CRMCall092016/ViewControllers/CustomerListViewController.swift"
+                  timestampString = "498119981.617965"
+                  startingColumnNumber = "9223372036854775807"
+                  endingColumnNumber = "9223372036854775807"
+                  startingLineNumber = "268"
+                  endingLineNumber = "268"
+                  offsetFromSymbolStart = "49">
+               </Location>
+               <Location
+                  shouldBeEnabled = "Yes"
+                  ignoreCount = "0"
+                  continueAfterRunningActions = "No"
+                  symbolName = "CRMCall092016.CustomerListViewController.(scrollRowToVisible (Swift.Int, animated : Swift.Bool) -&gt; ()).(implicit closure #1)"
+                  moduleName = "CRMCall092016"
+                  usesParentBreakpointCondition = "Yes"
+                  urlString = "file:///Users/hanbiro/Desktop/PG/CRM_CALL/CRMCall092016/CRMCall092016/ViewControllers/CustomerListViewController.swift"
+                  timestampString = "498119981.61947"
+                  startingColumnNumber = "9223372036854775807"
+                  endingColumnNumber = "9223372036854775807"
+                  startingLineNumber = "268"
+                  endingLineNumber = "268"
+                  offsetFromSymbolStart = "29">
+               </Location>
+            </Locations>
+         </BreakpointContent>
+      </BreakpointProxy>
+      <BreakpointProxy
+         BreakpointExtensionID = "Xcode.Breakpoint.FileBreakpoint">
+         <BreakpointContent
+            shouldBeEnabled = "No"
+            ignoreCount = "0"
+            continueAfterRunningActions = "No"
+            filePath = "CRMCall092016/ViewControllers/CustomerListViewController.swift"
+            timestampString = "498119043.005097"
+            startingColumnNumber = "9223372036854775807"
+            endingColumnNumber = "9223372036854775807"
+            startingLineNumber = "178"
+            endingLineNumber = "178"
             landmarkName = "searchCustomer()"
             landmarkType = "5">
          </BreakpointContent>
@@ -458,11 +570,11 @@
             ignoreCount = "0"
             continueAfterRunningActions = "No"
             filePath = "CRMCall092016/ViewControllers/CustomerListViewController.swift"
-            timestampString = "498044119.705732"
+            timestampString = "498119043.005097"
             startingColumnNumber = "9223372036854775807"
             endingColumnNumber = "9223372036854775807"
-            startingLineNumber = "149"
-            endingLineNumber = "149"
+            startingLineNumber = "209"
+            endingLineNumber = "209"
             landmarkName = "searchCustomer()"
             landmarkType = "5">
          </BreakpointContent>
diff --git a/CRMCall092016/Base.lproj/Main.storyboard b/CRMCall092016/Base.lproj/Main.storyboard
index 9e8dd86..47e77d0 100644
--- a/CRMCall092016/Base.lproj/Main.storyboard
+++ b/CRMCall092016/Base.lproj/Main.storyboard
@@ -1609,7 +1609,7 @@
                                     <font key="font" metaFont="system"/>
                                 </buttonCell>
                                 <connections>
-                                    <action selector="actionUnregister:" target="lma-nu-tEE" id="DJN-8c-EYD"/>
+                                    <action selector="actionUnregisters:" target="lma-nu-tEE" id="GmK-j9-CLK"/>
                                 </connections>
                             </button>
                             <button verticalHuggingPriority="750" translatesAutoresizingMaskIntoConstraints="NO" id="6On-oo-OcJ">
@@ -1912,6 +1912,127 @@ Gw
             </objects>
             <point key="canvasLocation" x="647" y="771"/>
         </scene>
+        <!--Window Controller-->
+        <scene sceneID="D7M-Hh-jI1">
+            <objects>
+                <windowController id="GT2-i4-fAC" customClass="StaffAvailabilityWindowController" customModule="CRMCall092016" customModuleProvider="target" sceneMemberID="viewController">
+                    <window key="window" identifier="StaffAvailabilityWindowController" title="Window" allowsToolTipsWhenApplicationIsInactive="NO" autorecalculatesKeyViewLoop="NO" oneShot="NO" releasedWhenClosed="NO" showsToolbarButton="NO" visibleAtLaunch="NO" frameAutosaveName="" animationBehavior="default" id="ywZ-4p-oi7">
+                        <windowStyleMask key="styleMask" titled="YES" closable="YES" miniaturizable="YES" resizable="YES"/>
+                        <windowPositionMask key="initialPositionMask" leftStrut="YES" rightStrut="YES" topStrut="YES" bottomStrut="YES"/>
+                        <rect key="contentRect" x="163" y="199" width="480" height="270"/>
+                        <rect key="screenRect" x="0.0" y="0.0" width="1280" height="777"/>
+                    </window>
+                    <connections>
+                        <segue destination="tyb-fH-Y4F" kind="relationship" relationship="window.shadowedContentViewController" id="pOj-E9-UZK"/>
+                    </connections>
+                </windowController>
+                <customObject id="D6D-qN-u9e" userLabel="First Responder" customClass="NSResponder" sceneMemberID="firstResponder"/>
+            </objects>
+            <point key="canvasLocation" x="688" y="2018"/>
+        </scene>
+        <!--Staff Availability View Controller-->
+        <scene sceneID="kKj-Fd-exS">
+            <objects>
+                <viewController storyboardIdentifier="StaffAvailabilityViewControllerID" id="tyb-fH-Y4F" customClass="StaffAvailabilityViewController" customModule="CRMCall092016" customModuleProvider="target" sceneMemberID="viewController">
+                    <view key="view" id="0rE-Uk-yHZ">
+                        <rect key="frame" x="0.0" y="0.0" width="480" height="482"/>
+                        <autoresizingMask key="autoresizingMask"/>
+                        <subviews>
+                            <scrollView fixedFrame="YES" autohidesScrollers="YES" horizontalLineScroll="26" horizontalPageScroll="10" verticalLineScroll="26" verticalPageScroll="10" usesPredominantAxisScrolling="NO" translatesAutoresizingMaskIntoConstraints="NO" id="tMN-wJ-jcg">
+                                <rect key="frame" x="87" y="48" width="272" height="400"/>
+                                <clipView key="contentView" ambiguous="YES" drawsBackground="NO" id="VeT-Mz-y8q">
+                                    <rect key="frame" x="1" y="1" width="270" height="398"/>
+                                    <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+                                    <subviews>
+                                        <outlineView verticalHuggingPriority="750" allowsExpansionToolTips="YES" columnAutoresizingStyle="lastColumnOnly" selectionHighlightStyle="sourceList" multipleSelection="NO" autosaveColumns="NO" rowHeight="24" rowSizeStyle="automatic" viewBased="YES" indentationPerLevel="16" outlineTableColumn="gRH-mZ-gKD" id="eKR-Fw-dby">
+                                            <rect key="frame" x="0.0" y="0.0" width="270" height="398"/>
+                                            <autoresizingMask key="autoresizingMask"/>
+                                            <size key="intercellSpacing" width="3" height="2"/>
+                                            <color key="backgroundColor" name="_sourceListBackgroundColor" catalog="System" colorSpace="catalog"/>
+                                            <color key="gridColor" name="gridColor" catalog="System" colorSpace="catalog"/>
+                                            <tableColumns>
+                                                <tableColumn width="267" minWidth="16" maxWidth="1000" id="gRH-mZ-gKD">
+                                                    <tableHeaderCell key="headerCell" lineBreakMode="truncatingTail" borderStyle="border">
+                                                        <font key="font" metaFont="smallSystem"/>
+                                                        <color key="textColor" name="headerTextColor" catalog="System" colorSpace="catalog"/>
+                                                        <color key="backgroundColor" name="headerColor" catalog="System" colorSpace="catalog"/>
+                                                    </tableHeaderCell>
+                                                    <textFieldCell key="dataCell" lineBreakMode="truncatingTail" selectable="YES" editable="YES" title="Text Cell" id="pq1-hw-xw8">
+                                                        <font key="font" metaFont="system"/>
+                                                        <color key="textColor" name="controlTextColor" catalog="System" colorSpace="catalog"/>
+                                                        <color key="backgroundColor" name="controlBackgroundColor" catalog="System" colorSpace="catalog"/>
+                                                    </textFieldCell>
+                                                    <tableColumnResizingMask key="resizingMask" resizeWithTable="YES" userResizable="YES"/>
+                                                    <prototypeCellViews>
+                                                        <tableCellView identifier="HeaderCell" id="9Ve-4B-3jR">
+                                                            <rect key="frame" x="1" y="1" width="267" height="17"/>
+                                                            <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+                                                            <subviews>
+                                                                <textField verticalHuggingPriority="750" fixedFrame="YES" translatesAutoresizingMaskIntoConstraints="NO" id="3fJ-5C-1kN">
+                                                                    <rect key="frame" x="0.0" y="1" width="145" height="14"/>
+                                                                    <textFieldCell key="cell" lineBreakMode="truncatingTail" sendsActionOnEndEditing="YES" title="HEADER CELL" id="ZIG-TX-Dhh">
+                                                                        <font key="font" metaFont="smallSystemBold"/>
+                                                                        <color key="textColor" name="headerColor" catalog="System" colorSpace="catalog"/>
+                                                                        <color key="backgroundColor" name="controlColor" catalog="System" colorSpace="catalog"/>
+                                                                    </textFieldCell>
+                                                                </textField>
+                                                            </subviews>
+                                                            <connections>
+                                                                <outlet property="textField" destination="3fJ-5C-1kN" id="dia-ld-kPy"/>
+                                                            </connections>
+                                                        </tableCellView>
+                                                        <tableCellView identifier="DataCell" id="tF0-CD-yNv">
+                                                            <rect key="frame" x="1" y="20" width="267" height="17"/>
+                                                            <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+                                                            <subviews>
+                                                                <imageView fixedFrame="YES" translatesAutoresizingMaskIntoConstraints="NO" id="130-zu-Hmp">
+                                                                    <rect key="frame" x="3" y="0.0" width="17" height="17"/>
+                                                                    <imageCell key="cell" refusesFirstResponder="YES" imageScaling="proportionallyDown" image="NSActionTemplate" id="Ve9-lx-3b3"/>
+                                                                </imageView>
+                                                                <textField verticalHuggingPriority="750" fixedFrame="YES" translatesAutoresizingMaskIntoConstraints="NO" id="agt-ml-G93">
+                                                                    <rect key="frame" x="25" y="0.0" width="120" height="17"/>
+                                                                    <textFieldCell key="cell" lineBreakMode="truncatingTail" sendsActionOnEndEditing="YES" title="Table View Cell" id="MnW-tD-FcL">
+                                                                        <font key="font" metaFont="system"/>
+                                                                        <color key="textColor" name="controlTextColor" catalog="System" colorSpace="catalog"/>
+                                                                        <color key="backgroundColor" name="controlColor" catalog="System" colorSpace="catalog"/>
+                                                                    </textFieldCell>
+                                                                </textField>
+                                                            </subviews>
+                                                            <connections>
+                                                                <outlet property="imageView" destination="130-zu-Hmp" id="ufq-e5-aqV"/>
+                                                                <outlet property="textField" destination="agt-ml-G93" id="Cbt-ZM-d3i"/>
+                                                            </connections>
+                                                        </tableCellView>
+                                                    </prototypeCellViews>
+                                                </tableColumn>
+                                            </tableColumns>
+                                            <connections>
+                                                <outlet property="dataSource" destination="tyb-fH-Y4F" id="sZm-xT-oQz"/>
+                                                <outlet property="delegate" destination="tyb-fH-Y4F" id="dAc-xz-iOG"/>
+                                            </connections>
+                                        </outlineView>
+                                    </subviews>
+                                    <nil key="backgroundColor"/>
+                                </clipView>
+                                <scroller key="horizontalScroller" hidden="YES" verticalHuggingPriority="750" horizontal="YES" id="GIj-uX-cj5">
+                                    <rect key="frame" x="1" y="119" width="238" height="15"/>
+                                    <autoresizingMask key="autoresizingMask"/>
+                                </scroller>
+                                <scroller key="verticalScroller" hidden="YES" verticalHuggingPriority="750" horizontal="NO" id="FTk-BX-msI">
+                                    <rect key="frame" x="224" y="17" width="15" height="102"/>
+                                    <autoresizingMask key="autoresizingMask"/>
+                                </scroller>
+                            </scrollView>
+                        </subviews>
+                    </view>
+                    <connections>
+                        <outlet property="sourceView" destination="eKR-Fw-dby" id="mFa-jU-7ib"/>
+                    </connections>
+                </viewController>
+                <customObject id="O4C-iu-X1T" userLabel="First Responder" customClass="NSResponder" sceneMemberID="firstResponder"/>
+            </objects>
+            <point key="canvasLocation" x="688" y="2530"/>
+        </scene>
         <!--View Controller-->
         <scene sceneID="hIz-AP-VOD">
             <objects>
@@ -2060,6 +2181,7 @@ Gw
     </scenes>
     <resources>
         <image name="Image" width="512" height="512"/>
+        <image name="NSActionTemplate" width="14" height="14"/>
         <image name="phoneIcon" width="50" height="50"/>
     </resources>
 </document>
diff --git a/CRMCall092016/Helpers/CRMCallHelpers.swift b/CRMCall092016/Helpers/CRMCallHelpers.swift
index d3bb409..b117a39 100644
--- a/CRMCall092016/Helpers/CRMCallHelpers.swift
+++ b/CRMCall092016/Helpers/CRMCallHelpers.swift
@@ -82,6 +82,7 @@ final class CRMCallHelpers {
     
     struct NameScreen {
         static let LoginWindowController = "LoginWindowController"
+        static let StaffAvailabilityWindowController = "StaffAvailabilityWindowController"
         static let HistoryCallWindowController = "HistoryCallWindowController"
         static let RingIngWindowController = "RingIngWindowController"
         static let CustomerListViewController = "CustomerListViewController"
diff --git a/CRMCall092016/ViewControllers/CustomerListViewController.swift b/CRMCall092016/ViewControllers/CustomerListViewController.swift
index 16391dc..2eceb9c 100644
--- a/CRMCall092016/ViewControllers/CustomerListViewController.swift
+++ b/CRMCall092016/ViewControllers/CustomerListViewController.swift
@@ -24,7 +24,9 @@ class CustomerListViewController: NSViewController, ViewControllerProtocol {
     
     private var dataDict = [[String: AnyObject]]()
     private var indexStart = 0
-    private var offset = 15
+    private var offset = 30
+    private var indexScroll = 0
+    private var isReplaceSearch = true
     
     private var itemSelect = [String: AnyObject]()
     
@@ -48,7 +50,12 @@ class CustomerListViewController: NSViewController, ViewControllerProtocol {
     }
 
     func configItems() {
-
+        
+        let clipView = self.tableViewCustomers.enclosingScrollView!.contentView
+        NSNotificationCenter.defaultCenter().addObserver(self,selector:#selector(CustomerListViewController.myBoundsChangeNotificationHandler(_:)),
+        name:NSViewBoundsDidChangeNotification,
+        object:clipView);
+        
     }
     // MARK: - View life cycle
     override func viewDidLoad() {
@@ -64,13 +71,18 @@ class CustomerListViewController: NSViewController, ViewControllerProtocol {
         
     }
     
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NSViewBoundsDidChangeNotification, object: nil)
+    }
+    
     // MARK: - Handing event
     
     @IBAction func actionSeach(sender: AnyObject) {
+        isReplaceSearch = true
         searchCustomer()
     }
     
-    @IBAction func actionUnregister(sender: AnyObject) {
+    @IBAction func actionUnregisters(sender: AnyObject) {
         delegate?.chooseCustomer(withDict: nil)
         
         dispatch_async(dispatch_get_main_queue(), {
@@ -115,6 +127,21 @@ class CustomerListViewController: NSViewController, ViewControllerProtocol {
             }
         })
     }
+    
+    func myBoundsChangeNotificationHandler(withNotification :NSNotification) {
+        let range = getVisibleRow()
+        let local = range.location + range.length
+
+        println("xxxxx \(local)")
+        if isReplaceSearch && local == (dataDict.count - 2) && indexScroll < local {
+            
+            indexScroll = local
+            isReplaceSearch = false
+            indexStart += (offset + 1)
+            
+            searchCustomer()
+        }
+    }
 
     // MARK: - SearchCustomer
     func searchCustomer() {
@@ -146,7 +173,10 @@ class CustomerListViewController: NSViewController, ViewControllerProtocol {
         
         AlamofireManager.requestUrlByGET(withURL: url, parameter: nil) { (datas, success) in
             if success {
-                self.dataDict.removeAll()
+                
+                if self.isReplaceSearch {
+                    self.dataDict.removeAll()
+                }
                 
                 println("-----------> SEARCH CUSTOMER DATA CALL RESPONCE: \(datas)")
                 
@@ -177,6 +207,7 @@ class CustomerListViewController: NSViewController, ViewControllerProtocol {
                     }
                     
                     self.tableViewCustomers.reloadData()
+                    self.isReplaceSearch = true
                 } else {
                     println("Not found SEARCH CUSTOMER from server")
                 }
@@ -221,5 +252,34 @@ extension CustomerListViewController: NSTableViewDelegate, NSTableViewDataSource
         }
     }
     
+    func getVisibleRow() -> NSRange {
+        if let scrollView = self.tableViewCustomers.enclosingScrollView {
+            let visibleRect = scrollView.contentView.visibleRect
+            let range = self.tableViewCustomers.rowsInRect(visibleRect)
+            return range
+        }
+        return NSRange()
+    }
     
+    func scrollRowToVisible(row: Int) {
+        println("\(row)")
+    }
+    func scrollRowToVisible(row: Int, animated: Bool) {
+        println("\(row)")
+    }
+//    func tableView(tableView: NSTableView, willDisplayCell cell: AnyObject, forTableColumn tableColumn: NSTableColumn?, row: Int) {
+//       // println("\(row)")
+//        
+//        let range = getVisibleRow()
+//        let local = range.location + range.length
+//        println("loacal \(range.location) late row \(range.length)")
+//        
+//        if isReplaceSearch && local == (dataDict.count - 5) {
+//            
+//            isReplaceSearch = false
+//            indexStart += (offset + 1)
+//            
+//            searchCustomer()
+//        }
+//    }
 }
diff --git a/CRMCall092016/ViewControllers/HistoryCallViewController.swift b/CRMCall092016/ViewControllers/HistoryCallViewController.swift
index c146263..09cf986 100644
--- a/CRMCall092016/ViewControllers/HistoryCallViewController.swift
+++ b/CRMCall092016/ViewControllers/HistoryCallViewController.swift
@@ -776,8 +776,23 @@ extension HistoryCallViewController: CustomerListDelegate {
             customerSelect = d
             
             if d["type"] as! String == CRMCallHelpers.TypeApi.Employee.rawValue {
+                let userInfo = UserInfo()
+                
                 if let name = d["name"] as? String {
                     self.nameTextField.stringValue = name
+                    userInfo.name = name
+                }
+                
+                if let code = d["code"] as? String {
+                    userInfo.code = code
+                }
+                
+                if let phone = d["phone"] as? String {
+                    userInfo.phone = phone
+                }
+                
+                if let parentName = d["parentName"] as? String {
+                    userInfo.parentName = parentName
                 }
                 
                 if let name = d["company"] as? String {
@@ -786,6 +801,8 @@ extension HistoryCallViewController: CustomerListDelegate {
                 
                 self.nameTextField.enabled = false
                 self.companyTextField.enabled = false
+                
+                self.customerDict = CRMCallHelpers.createDictionaryCustomer(withData: userInfo)
             }
             
             if d["type"] as! String == CRMCallHelpers.TypeApi.Company.rawValue {
diff --git a/CRMCall092016/ViewControllers/MainViewController.swift b/CRMCall092016/ViewControllers/MainViewController.swift
index f55f41c..1bfa419 100644
--- a/CRMCall092016/ViewControllers/MainViewController.swift
+++ b/CRMCall092016/ViewControllers/MainViewController.swift
@@ -435,12 +435,20 @@ class MainViewController: NSViewController  , ViewControllerProtocol{
         //        })
         
         // dispatch_async(dispatch_get_main_queue(), {
-        if let historyWindowController = CRMCallManager.shareInstance.screenManager[CRMCallHelpers.NameScreen.HistoryCallWindowController] {
-            historyWindowController.showWindow(nil)
+//        if let historyWindowController = CRMCallManager.shareInstance.screenManager[CRMCallHelpers.NameScreen.HistoryCallWindowController] {
+//            historyWindowController.showWindow(nil)
+//        } else {
+//            let historyWindowController = HistoryCallWindowController.createInstance()
+//            historyWindowController.showWindow(nil)
+//            CRMCallManager.shareInstance.screenManager[CRMCallHelpers.NameScreen.HistoryCallWindowController] = historyWindowController
+//        }
+        
+        if let staffAvailibilityWindowController = CRMCallManager.shareInstance.screenManager[CRMCallHelpers.NameScreen.StaffAvailabilityWindowController] {
+            staffAvailibilityWindowController.showWindow(nil)
         } else {
-            let historyWindowController = HistoryCallWindowController.createInstance()
-            historyWindowController.showWindow(nil)
-            CRMCallManager.shareInstance.screenManager[CRMCallHelpers.NameScreen.HistoryCallWindowController] = historyWindowController
+            let staffAvailibilityWindowController = StaffAvailabilityWindowController.createInstance()
+            staffAvailibilityWindowController.showWindow(nil)
+            CRMCallManager.shareInstance.screenManager[CRMCallHelpers.NameScreen.HistoryCallWindowController] = staffAvailibilityWindowController
         }
         
         //   })
diff --git a/CRMCall092016/ViewControllers/StaffAvailabilityViewController.swift b/CRMCall092016/ViewControllers/StaffAvailabilityViewController.swift
new file mode 100644
index 0000000..e06fb23
--- /dev/null
+++ b/CRMCall092016/ViewControllers/StaffAvailabilityViewController.swift
@@ -0,0 +1,57 @@
+//
+//  StaffAvailabilityViewController.swift
+//  CRMCall092016
+//
+//  Created by Hanbiro on 10/14/16.
+//  Copyright © 2016 xuanvinhtd. All rights reserved.
+//
+
+import Cocoa
+
+class StaffAvailabilityViewController: NSViewController, ViewControllerProtocol {
+    
+    // MARK: - Properties
+    @IBOutlet weak var sourceView: NSOutlineView!
+    
+    // MARK: - Initialzation
+    static func createInstance() -> NSViewController {
+        return  CRMCallHelpers.storyBoard.instantiateControllerWithIdentifier("StaffAvailabilityViewControllerID") as! StaffAvailabilityViewController
+    }
+    
+    func initData() {
+        ///-------------- GET ALL STAFF -------------//
+        let url = CRMCallConfig.API.getAllStaffs()
+        
+        AlamofireManager.requestUrlByGET(withURL: url, parameter: nil) { (datas, success) in
+            if success {
+                println("-----------> Get All Staff data responce: \(datas)")
+                
+                //                guard let data = datas["rows"] as? [String: AnyObject] else {
+                //                    println("Cannot get data after register employee success")
+                //                    return
+                //                }
+            } else {
+                println("---XXXXX---->>> Get all staff data fail with message: \(datas)")
+            }
+        }
+
+    }
+    
+    // MARK: - View life Cycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        initData()
+        // Do view setup here.
+    }
+    
+}
+// MARK: - Outline View Data Source
+extension StaffAvailabilityViewController: NSOutlineViewDataSource {
+    
+}
+
+// MARK: - Outline View Data Delegate
+extension StaffAvailabilityViewController: NSOutlineViewDelegate {
+    
+}
diff --git a/CRMCall092016/WindowController/StaffAvailabilityWindowController.swift b/CRMCall092016/WindowController/StaffAvailabilityWindowController.swift
new file mode 100644
index 0000000..666b86b
--- /dev/null
+++ b/CRMCall092016/WindowController/StaffAvailabilityWindowController.swift
@@ -0,0 +1,24 @@
+//
+//  StaffAvailabilityWindowController.swift
+//  CRMCall092016
+//
+//  Created by Hanbiro on 10/14/16.
+//  Copyright © 2016 xuanvinhtd. All rights reserved.
+//
+
+import Cocoa
+
+class StaffAvailabilityWindowController: NSWindowController {
+    
+    // MARK: - Initialzation
+    static func createInstance() -> NSWindowController {
+        return  CRMCallHelpers.storyBoard.instantiateControllerWithIdentifier("StaffAvailabilityWindowControllerID") as! StaffAvailabilityWindowController
+    }
+    
+    override func windowDidLoad() {
+        super.windowDidLoad()
+        
+        // Implement this method to handle any initialization after your window controller's window has been loaded from its nib file.
+    }
+    
+}
\ No newline at end of file
